jwst/regtest/test_associations_sdp_pools.py

```diff
@@ -1,251 +1,158 @@
-    },
-    'jw04225_20241213t150701DMS_pool': {
-        'args': ['-i', 'o001', 'o002', '--DMS'],
-        'xfail': None,
-        'slow': False,
-    },
-    'jw80600_20171108T041522_pool': {
-        'args': [],
-        'xfail': 'PR #3450',
-        'slow': False,
-    },
-    'jw82600_20180921T023255_pool': {
-        'args': [],
-        'xfail': None,
-        'slow': True
-    },
-    'jw93065_20171108T041402_pool': {
-        'args': [],
-        'xfail': None,
-        'slow': True,
-    },
-    'jw93135_20171108T041617_pool': {
-        'args': [],
-        'xfail': None,
-        'slow': True,
-    },
-    'jw93135_20171108T041617-fixed_pool': {
-        'args': [],
-        'xfail': None,
-        'slow': True,
-    },
-    'jw94015_20171108T041516_pool': {
-        'args': [],
-        'xfail': None,
-        'slow': True,
-    },
-    'jw98010_20171108T062332_pool': {
-        'args': [],
-        'xfail': 'PR #3450',
-        'slow': False,
-    },
-}
-
-
-# #####
-# Tests
-# #####
-@pytest.mark.filterwarnings('error')
-def test_against_standard(sdpdata_module, pool_path, slow):
-    """Compare a generated association against a standard
-
-    Success is when no other AssertionError occurs.
-    """
+pool_regex = re.compile(r"(?P<proposal>jw.+?)_(?P<versionid>.+)_pool")
 
-    # Parse pool name
-    pool = Path(pool_path).stem
-    proposal, version_id = pool_regex.match(pool).group('proposal', 'versionid')
-    special = SPECIAL_POOLS.get(pool, SPECIAL_DEFAULT)
-
-    if special['slow'] and not slow:
-        pytest.skip(f'Pool {pool} requires "--slow" option')
-
-    # Setup test path
-    cwd = Path(pool)
-    cwd.mkdir()
-    with pushdir(cwd):
-
-        # Create the generator running arguments
-        output_path = Path(pool)
-        output_path.mkdir()
-        sdpdata_module.output = str(output_path)
-        args = special['args'] + [
-            '-p', sdpdata_module.output,
-            '--version-id', version_id,
-            sdpdata_module.get_data(pool_path)
-        ]
-
-        # Create the associations
-        asn_generate.cli(
-            args)
-
-        # Compare to the truth associations.
-        truth_paths = sdpdata_module.truth_paths(pool)
-        try:
-            compare_asn_files(output_path.glob('*.json'), truth_paths)
-        except AssertionError:
-            if special['xfail']:
-                pytest.xfail(special['xfail'])
-            else:
-                raise
+
+def parfunc(x):
+    """
+    For readable param ids.
+    ``x`` is tuple of ``(pool, args)``.
+    """
+    return x[0]
+
+
+def _assoc_sdp_against_standard(rtdata, resource_tracker, pool_args):
+    pool, args = pool_args
+    proposal, version_id = pool_regex.match(pool).group(
+        "proposal", "versionid")
+
+    input_csv = rtdata.get_data(f"associations/sdp/pools/{pool}.csv")
+    rtdata.output = os.curdir   # This test is jailed and parametrized
+    # Create the associations
+    with resource_tracker.track():
```

Comment:
I like having tracking for these tests - I think the implementation needs a tweak though, following the example shown here: https://github.com/spacetelescope/ci_watson/blob/eeb0fd74915ad9ed390eee5fa628df5294273781/ci_watson/resource_tracker.py#L10

I confirmed that the tracking isn't logged in the current implementation - I don't see any logging in the results xml file from the RT run.

Reply:
Oh OK thanks! I copied this from a random example I found elsewhere in this repo. 😬